게시글 CRUD api (회원 없이 ver) #51

bjho606 opened the pull request:
## 📄 Summary
- #51 

<br>

## 🙋🏻 To Reviewers
- merge 할 때 이슈51 close 해야합니다.
- 회원 추가해서 더 완성해야합니다.

bjho606 committed:
feat: 게시글 API 구현 (1차 완성)

- 게시글 CRUD 기능 구현
	- 게시글 생성
	- 게시글 상세내용 불러오기
	- 게시판의 모든 게시글 목록 불러오기
	- 게시글 제목,내용 수정하기
	- 게시글 숨기기/안숨기기(같은 api로 숨김처리/활성화처리)
	- 숨김 게시글 목록 불러오기
	- 게시글의 게시판 변경하기
	- 게시글 삭제하기
- swagger api 명세서 작성

<할 일>
- Member 테이블과 연결시키기 (최종 완성)
- 게시글 목록 pagination 적용

bjho606 committed:
feat: 게시글 목록 불러오기에 정렬 + 페이지네이션 적용

chwangmin commented on src/main/java/com/ssafy/naem/domain/feed/service/FeedService.java:
```diff
@@ -0,0 +1,122 @@
+import java.util.Optional;
+
+@Service
+public class FeedService {
+
+    private final FeedRepository feedRepository;
+    private final BoardRepository boardRepository;
+
+    @Autowired
+    public FeedService(FeedRepository feedRepository, BoardRepository boardRepository) {
+        this.feedRepository = feedRepository;
+        this.boardRepository = boardRepository;
+    }
```
이 부분 에서
DI 를 할 때, @RequiredArgsConstructor 이걸 써서 구현하지 않은 이유가 있을까요?

생성자 코드를 추가하지 않고, 코드를 줄일 수 있어 좋은 방법이라고 생각해서 여쭤봤습니다!

bjho606 replied:
반영했습니다!

chwangmin commented on src/main/java/com/ssafy/naem/domain/board/entity/Board.java:
```diff
@@ -15,17 +17,24 @@ public class Board {
     @Builder.Default
     private Status status = Status.STATUS_ACTIVE;
 
+    @OneToMany(mappedBy = "board")
+    List<Feed> feedList;
+
```
다대일 양방향 좋습니다.

chwangmin reviewed (approved):
코드 깔끔하게 구현하셨네요! 고생하셨습니다.

minju412 reviewed (approved):
고생하셨습니다~!! 👍👍

bjho606 commented on src/main/java/com/ssafy/naem/domain/feed/controller/FeedController.java:
```diff
@@ -0,0 +1,118 @@
+    }
+
+    @ResponseBody
+    @PatchMapping("/{id}/delete")
```
네 맞습니다!

bjho606 commented on src/main/java/com/ssafy/naem/domain/feed/controller/FeedController.java:
```diff
@@ -0,0 +1,118 @@
+    @ResponseBody
+    @GetMapping("/{id}")
+    @Operation(summary = "Get Feed", description = "게시글의 상세 내용을 불러온다.")
+    public BaseResponse<FeedResponse> getFeed(@PathVariable("id") Long id) {
```
GetMapping 어노테이션으로 어떤 인자가 들어가는지 쉽게 보여서 다른 함수들과의 통일성 측면에서 낫지 않을까요?

bjho606 committed:
refactor: PR 요청 반영

- @requiredargsconstructor로 생성자 코드 줄이기
- 게시글의 게시판 정보 변경 내용을 게시글을 수정할 때, 한꺼번에 처리하도록 리팩토링
- 한 게시판의 모든 게시글 내용을 불러올 때, boardId를 request body -> query string으로 받도록 변경 (필터링의 개념이므로)

bjho606 commented on src/main/java/com/ssafy/naem/domain/feed/controller/FeedController.java:
```diff
@@ -0,0 +1,118 @@
+    }
+
+    @ResponseBody
+    @PatchMapping("/{id}/board")
```
게시글 내용을 변경할 때 아예 게시판 정보도 변경하도록 리팩토링했습니다!
